library/pool.c

```diff
@@ -55,7 +55,10 @@ static void dnet_work_pool_stop(struct dnet_work_pool_place *place)
 
 	for (i = 0; i < place->pool->num; ++i) {
 		wio = &place->pool->wio_list[i];
-		pthread_join(wio->tid, NULL);
+		if (!wio->joined) {
+			pthread_join(wio->tid, NULL);
+			wio->joined = 1;
+		}
```

Comment:
This looks like a bug, previously pool was deleted first and never tried to be deleted again.
Pool has to be removed first before caches and backends, since pool can grab io requests from the queue and start processing them, if you've removed cache, it will crash. This is why I asked about order of `dnet_node_cleanup_common_resources()` call which you changed and which looks like a bug.